code/API_definitions/connected-network-type.yaml

```diff
@@ -0,0 +1,416 @@
+      security:
+        - openId:
+            - connected-network-type:read
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/RequestConnectedNetworkType"
+        required: true
+      responses:
+        "200":
+          description: Contains connected network type
+          headers:
+            x-correlator:
+              $ref: '#/components/headers/x-correlator'
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ConnectedNetworkTypeResponse"
+              examples:
+                Connected-to-4GLTE:
+                  value:
+                    lastStatusTime: "2024-02-20T10:41:38.657Z"
+                    connectedNetworkType: 4GLTE
+                Connected-To-5GNSA:
+                  value:
+                    lastStatusTime: "2024-02-20T10:41:38.657Z"
+                    connectedNetworkType: 5GNSA
+                Connected-To-Non3gpp4g:
+                  value:
+                    lastStatusTime: "2024-02-20T10:41:38.657Z"
+                    connectedNetworkType: NON3GPP4G
+                Not-Connected:
+                  value:
+                    lastStatusTime: "2024-02-20T10:41:38.657Z"
+                    connectedNetworkType: NOT_CONNECTED
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/ConnectedNetworkTypePermissionDenied403"
+        "404":
+          $ref: "#/components/responses/ConnectedStatusNotFound404"
+        "422":
+          $ref: "#/components/responses/ConnectedStatusUnprocessableEntity422"
+        "500":
+          $ref: "#/components/responses/Generic500"
+        "503":
+          $ref: "#/components/responses/Generic503"
+components:
+  securitySchemes:
+    openId:
+      type: openIdConnect
+      openIdConnectUrl: https://example.com/.well-known/openid-configuration
+  parameters:
+    x-correlator:
+      name: x-correlator
+      in: header
+      description: Correlation id for the different services
+      schema:
+        type: string
+  headers:
+    x-correlator:
+      description: Correlation id for the different services
+      schema:
+        type: string
+  schemas:
+    LastStatusTime:
+      description: Last time that the associated connected Network Type was updated
+      type: string
+      format: date-time
+      example: "2024-02-20T10:41:38.657Z"
+
+    ConnectedNetworkTypeResponse:
+      type: object
+      required:
+        - connectedNetworkType
+      properties:
+        lastStatusTime:
+          $ref: "#/components/schemas/LastStatusTime"
+        connectedNetworkType:
+          $ref: "#/components/schemas/ConnectedNetworkType"
+
+    ConnectedNetworkType:
+      description: |
+        * NOT_CONNECTED: The device is not connected to network
```

Comment:
I disagree with using `NOT_CONNECTED`. It should be `UNKOWN` (or `NOT_KNOWN`, or similar).

There are two reasons for this:
- There are various reasons why the API provider might not know the network type, and not being connected to the network at all is just one of them. Rather than have separate codes for all these reasons, and the API provider trying to work out what the reason is, better just to say "I don't know".
- The purpose of the API is to return the network type. Saying the device is not connected is providing information over and above this and, indeed, is information that is anyway provided by a different API (`device-reachability-status`). We should avoid having two APIs that do the same thing in different ways.

If there is an insistence that this API also returns information as to whether or not the device is connected to the network, then it should be an extension of `device-reachability-status` and not a new API.

Reply:
Updated to UNKNOWN as suggested